src/accordion/accordion.directive.ts

```diff
@@ -0,0 +1,305 @@
+		'[class.accordion-header]': 'true',
+	},
+})
+export class NgbAccordionHeader {
+	constructor() {}
+}
+
+@Directive({
+	selector: '[ngbAccordionItem]',
+	standalone: true,
+	host: {
+		'[class.accordion-item]': 'true',
+		'[id]': 'id',
+	},
+})
+export class NgbAccordionItem implements AfterContentInit {
+	constructor(
+		@Inject(forwardRef(() => NgbAccordionDirective)) private accordion: NgbAccordionDirective,
+		private cd: ChangeDetectorRef,
+	) {}
+
+	private subscriptions: Subscription[] = [];
+	private _collapsed = true;
+	private _id = `${nextId++}`;
+	private _disabled = false;
+
+	animatingBodyCollapse = false;
+
+	@ContentChild(NgbAccordionCollapse, { static: true }) private collapse: NgbAccordionCollapse;
+
+	@Input('ngbAccordionItem') set id(inputId: string) {
+		if (inputId) {
+			this._id = inputId;
+		}
+	}
+	@Input() destroyOnHide = this.accordion.destroyOnHide;
+	@Input() set disabled(isDisabled: boolean) {
+		if (isDisabled != this.disabled) {
+			this._disabled = isDisabled;
+			if (this._disabled) {
+				this._collapsed = true;
+				this.collapse.ngbCollapse.collapsed = true;
+			}
+		}
+	}
+	@Input() set collapsed(collapsed: boolean) {
+		if (this.disabled) {
+			if (!this.collapsed) {
+				this._collapsed = true;
+			}
+		} else if (this.collapsed !== collapsed) {
+			this._collapsed = collapsed;
+			// we need force CD to get template into DOM before starting animation to calculate its height correctly
+			if (!this._collapsed) {
+				this.animatingBodyCollapse = true;
+				this.cd.detectChanges();
+			}
+			// we also need to make sure 'animation' flag is up-to- date
+			this.collapse.ngbCollapse.animation = this.accordion.animation;
+			this.collapse.ngbCollapse.collapsed = collapsed;
+		}
+	}
+
+	@Output() shown = new EventEmitter<void>();
+	@Output() hidden = new EventEmitter<void>();
+
+	get collapsed(): boolean {
+		return this._collapsed;
+	}
+	get disabled() {
+		return this._disabled;
+	}
+	public get id() {
+		return `ngb-accordion-item-${this._id}`;
+	}
+
+	get toggleId() {
+		return `ngb-accordion-toggle-${this._id}`;
+	}
+	get collapseId() {
+		return `ngb-accordion-collapse-${this._id}`;
+	}
+
+	ngAfterContentInit(): void {
+		const { ngbCollapse } = this.collapse;
+		// we need to disable the animation for the first init
+		ngbCollapse.animation = false;
+		ngbCollapse.collapsed = this.collapsed;
+		// we set the animation to the default of the accordion
+		ngbCollapse.animation = this.accordion.animation;
+		// event forwarding from 'ngbCollapse' to 'ngbAccordion'
+		this.subscriptions.push(
+			ngbCollapse.hidden.subscribe(() => {
+				// when the animation finishes we can remove the template from DOM
+				this.animatingBodyCollapse = false;
+				this.hidden.emit();
+				this.accordion.hidden.emit(this.id);
+			}),
+			ngbCollapse.shown.subscribe(() => {
+				this.shown.emit();
+				this.accordion.shown.emit(this.id);
+			}),
+		);
+	}
+
+	toggle() {
+		this.collapsed = !this.collapsed;
```

Comment:
Hmm, what about the fact is can be disabled? You're doing quite a lot of checks in `set collapse()`, but not here?